### PR TITLE
Frontend signup login

### LINKED ISSUE #41: Create the user login and signup page
Create a page for user sign up and login.
- [x] Form for signing up
- [x] Form for logging in
- [x] Session management
   - [x] Persist login over page refresh
   - [x] Session timeout after some period of inactivity
- [x] Unit tests with Vitest
- [x] End-to-end tests with Cypress

### PR DESCRIPTION
**Pull request by JuhoBjn**
Add a page to sign up and log in users. Login is persisted over page refreshes. Session times out after two hours of inactivity.

Alos updated backend login and signup endpoints to return error responses in JSON format.

Closes #41

### CODE REVIEW

**Review by vaasuu — APPROVED** [review]
Looks good(ish) and works ok. Site definitely isn't responsive yet, but maybe we'll fix that later. I left some comments and suggestions.

File: frontend/.eslintrc.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "react-refresh/only-export-components": [
       "warn",
       { "allowConstantExport": true }
-    ]
+    ],
+    "react/prop-types": [0]
```
[the remark this answers is not in the document]
Comment (JuhoBjn) [reply frontend/.eslintrc.json:21]: This is intentional. ESlint kept complaining about not checking prop types. Figured we wouldn't need to check prop types in this project, but I can add it if you want.

File: frontend/cypress/support/commands.js
[the remark this answers is not in the document]
Comment (JuhoBjn) [reply frontend/cypress/support/commands.js:1]: Refactored the tests. I wanted to keep checking the success of signing up and loggin in in the commands, so I added optional parameters to the functions for whether they should fail.

File: frontend/src/index.css
[the remark this answers is not in the document]
Comment (JuhoBjn) [reply frontend/src/index.css:1]: Thanks! 🙌

**Review by vaasuu — APPROVED** [review]
LGTM, but noticed that Cypress coverage generates files that overwrite some of the ones generated by vitest, it could be a problem. It's still possible to view the html report for both in the coverage directory.

**Pull request merged**